src/serde_macros.rs

```diff
@@ -96,7 +96,7 @@ pub mod serde_details {
         /// serde serialization
         fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
             if s.is_human_readable() {
-                s.serialize_str(&self.to_hex())
+                s.collect_str(self)
```

Comment:
I think it's probably generally useful to use `Display` for human-readable encodings, regardless of what they are.

Though I realize that your point is that we're using `from_hex` rather than `from_str` in our Deserialize impl, which means that this won't actually work except for hex-encoding types. Let me fix that.